Description: Move the cmake files from demo to project root and add install targets.

This makes it possible to use these for building under Linux and make a Debian package of the source.

Author: Marius Kotsbak <marius@kotsbak.com>
Bug-Debian: http://bugs.debian.org/692563

--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,32 @@
+
+CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
+
+PROJECT(RCF)
+
+SET(RCF_VERSION 1.3.1)
+
+SET(RCF_ROOT ${CMAKE_SOURCE_DIR})
+
+SET(
+    RCF_DEFINES
+    -DWIN32_LEAN_AND_MEAN
+    -DBOOST_ALL_NO_LIB
+    -DRCF_USE_BOOST_ASIO
+    -DRCF_USE_OPENSSL
+    -DRCF_USE_ZLIB
+    -DRCF_USE_BOOST_SERIALIZATION)
+
+SET(
+    RCF_INCLUDES 
+    ${RCF_ROOT}/include)
+
+ADD_SUBDIRECTORY(RcfLib)
+ADD_SUBDIRECTORY(RcfDll)
+#ADD_SUBDIRECTORY(DemoClient)
+#ADD_SUBDIRECTORY(DemoServer)
+
+INSTALL(DIRECTORY include/ DESTINATION include)
+
+SET_TARGET_PROPERTIES(rcf-cpp rcf-cpp-static
+  PROPERTIES
+    SOVERSION ${RCF_VERSION})
--- /dev/null
+++ b/RcfLib/CMakeLists.txt
@@ -0,0 +1,32 @@
+
+ADD_DEFINITIONS( ${RCF_DEFINES} )
+
+INCLUDE_DIRECTORIES( ${RCF_INCLUDES} )
+
+SET(LIBRARY_OUTPUT_PATH ${BIN_PATH})
+
+ADD_LIBRARY(
+    rcf-cpp-static
+    STATIC 
+    ${RCF_ROOT}/src/RCF/RCF.cpp)
+
+SET(Boost_USE_STATIC_LIBS ON)
+SET(Boost_USE_STATIC_RUNTIME ON)
+find_package( Boost 1.33.0 COMPONENTS serialization system REQUIRED)
+TARGET_LINK_LIBRARIES( rcf-cpp-static ${Boost_LIBRARIES} )
+
+find_package(Threads REQUIRED)
+target_link_libraries( rcf-cpp-static ${CMAKE_THREAD_LIBS_INIT} )
+
+find_package(OpenSSL REQUIRED)
+target_link_libraries( rcf-cpp-static ${OPENSSL_LIBRARIES} )
+
+find_package(ZLIB REQUIRED)
+target_link_libraries( rcf-cpp-static ${ZLIB_LIBRARIES} )
+
+INSTALL(TARGETS rcf-cpp-static
+  ARCHIVE DESTINATION lib RENAME rcf-cpp)
+
+SET_TARGET_PROPERTIES(rcf-cpp-static
+  PROPERTIES
+    OUTPUT_NAME rcf-cpp)
--- /dev/null
+++ b/RcfDll/CMakeLists.txt
@@ -0,0 +1,36 @@
+
+ADD_DEFINITIONS(
+    -DRCF_BUILD_DLL 
+    -DRCF_NO_AUTO_INIT_DEINIT)
+    
+ADD_DEFINITIONS( ${RCF_DEFINES} )
+
+# Disable some warnings
+# warning C4275: non dll-interface class '' used as base for dll-interface class ''
+# warning C4251: '' : class '' needs to have dll-interface to be used by clients of class ''
+IF(MSVC)
+ADD_DEFINITIONS(/wd4275 /wd4251)
+ENDIF(MSVC)
+
+INCLUDE_DIRECTORIES( ${RCF_INCLUDES} )
+
+ADD_LIBRARY(
+    rcf-cpp
+    SHARED 
+    ${RCF_ROOT}/src/RCF/RCF.cpp)
+
+SET(Boost_USE_STATIC_RUNTIME OFF)
+find_package( Boost 1.33.0 COMPONENTS serialization system REQUIRED)
+TARGET_LINK_LIBRARIES( rcf-cpp ${Boost_LIBRARIES} )
+
+find_package(Threads REQUIRED)
+target_link_libraries( rcf-cpp ${CMAKE_THREAD_LIBS_INIT} )
+
+find_package(OpenSSL REQUIRED)
+target_link_libraries( rcf-cpp ${OPENSSL_LIBRARIES} )
+
+find_package(ZLIB REQUIRED)
+target_link_libraries( rcf-cpp ${ZLIB_LIBRARIES} )
+
+INSTALL(TARGETS rcf-cpp
+  LIBRARY DESTINATION lib)
